torch/_dynamo/variables/misc.py

```diff
@@ -456,6 +457,15 @@ def call_method(
             return self.call_apply(tx, args, kwargs)
         elif name == "backward":
             with tx.strict_translation_mode():
+                # backward function can be decorated by torch.autograd.function.once_differentiable,
+                # however, torch.compile doesn't support higher order gradients. And we can assume that
+                # the grad_mode during the backwards pass is False, so just inline the original function.
+                if hasattr(
+                    self.fn_cls.backward, "__wrapped__"
+                ) and self.fn_cls.backward.__code__.co_filename == module_dir(
+                    torch.autograd.function
+                ):
+                    self.fn_cls.backward = self.fn_cls.backward.__wrapped__
```

Comment:
This is better, but:
Why are we replacing self.fn_cls.backward with `self.fn_cls.backward.__wrapped__` ? We'd like to inline through once_differentiable wrapping the backward, not just the backward

Reply:
> once_differentiable has different behavior depending on if the gradients have requires_grad=True or not. In theory we don't actually know if the gradients have requires_grad=True or not at the time of the forward pass, but we may actually be able to assume they don't because the rest of torch.compile assumes that...

I was trying to inline the ```once_differentiable``` wrapped function, but we can't guarantee it passes L572 in Dynamo tracing. Then I found actually ```once_differentiable``` wrapped function just does L570 actually, which is the same as inlining the original backward function.
https://github.com/pytorch/pytorch/blob/9a9232956f90b0adf53049badf91a4309fb29a87/torch/autograd/function.py#L566-L573

Reply:
We can't access ```grad.requires_grad``` in Dynamo tracing.

Reply:
The HOP tracing of the autograd.Function backward should set torch.is_grad_enabled to False, because PT2 doesn't support double backwards. It doesn't do this (because there is a bug).

Reply:
Ah, you are right! I think we should figure out why grad is enabled during Dynamo tracing backward graph.

Reply:
You want to do something like the following https://github.com/pytorch/pytorch/pull/108686/files#diff-d6c3a57794a9a8a79eacedc57d68ff631fe80776f8701398eb3ef349973b033bR971-R977 (this was my original once_differentiable PR that got reverted :P)